core/src/main/java/org/apache/accumulo/core/dataImpl/InstanceInfo.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.core.dataImpl;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Objects;
+
+import org.apache.accumulo.core.data.InstanceId;
+
+public class InstanceInfo {
+
+  private final String name;
+  private final InstanceId id;
```

Comment:
I made a comment elsewhere about just serializing and deserializing this object using a single file in HDFS in a known location. If we do that, then I think we can collapse the instance version into this object as well and get rid of that file. It may also make sense to add a version number to this object to handle changes over time, much like a serialVersionUid.

Reply:
Because of the goal to minimize the amount of cluster RPC requests needed to read the contents of the file, and only require talking to the HDFS NameNode, the design is to keep the info in the file names. However, I do think we could move the version into this instance directory with a different prefix, and also add it to this object.